Arrowgene.Ddon.GameServer/Characters/CharacterManager.cs

```diff
@@ -151,26 +152,41 @@ public void UpdateDatabaseOnExit(Character character)
             }
         }
 
-        public void UpdateCharacterExtendedParamsNtc(GameClient client, Character character)
+        public void UpdateCharacterExtendedParamsNtc(GameClient client, CharacterCommon character)
         {
             UpdateCharacterExtendedParams(character);
             NotifyClientOfCharacterStatus(client, character);
         }
 
-        private void NotifyClientOfCharacterStatus(GameClient client, Character character)
+        private void NotifyClientOfCharacterStatus(GameClient client, CharacterCommon character)
         {
-            S2CContextGetLobbyPlayerContextNtc ntc1 = new S2CContextGetLobbyPlayerContextNtc();
-            GameStructure.S2CContextGetLobbyPlayerContextNtc(ntc1, character);
-            client.Send(ntc1);
 
-            S2CExtendEquipSlotNtc ntc2 = new S2CExtendEquipSlotNtc()
+            if (character is Character)
             {
-                EquipSlot = EquipCategory.Jewelry,
-                AddNum = 0,
-                TotalNum = character.JewelrySlotNum
-            };
+                S2CContextGetLobbyPlayerContextNtc ntc1 = new S2CContextGetLobbyPlayerContextNtc();
+                GameStructure.S2CContextGetLobbyPlayerContextNtc(ntc1, (Character) character);
+                client.Send(ntc1);
 
-            client.Send(ntc2);
+                S2CExtendEquipSlotNtc ntc2 = new S2CExtendEquipSlotNtc()
+                {
+                    EquipSlot = EquipCategory.Jewelry,
+                    AddNum = 0,
+                    TotalNum = character.JewelrySlotNum
+                };
+
+                client.Send(ntc2);
+            }
+            else
+            {
+                PartyMember partyMember = client.Party.GetPartyMemberByCharacter(character);
```

Comment:
Do Pawns HAVE to be in a party when notifying? You could get the pawn instance from client.Pawns

Reply:
Hmm, at least for the dragon force upgrade, you can only trigger the upgrades with them being summoned into your party. What scenarios would someone ask for a pawn to be updated but not in the characters party? Does DDON allow you summon other players pawns like DD2?

Reply:
Yes, you can summon other players pawns, or be in a party where other player summoned their pawns. Either way, this function has a more "general" function, which is what its name says: NotifyClientOfCharacterStatus. It doesn't imply that the pawn has to be in your party or that it should only be used in the context of upgrading the BO tree. Considering you can as easily obtain the pawn instance without party, i think it'd be good to not have that dependency.

Reply:
The problem is that for the DF tree, I need the party instance to update the pawn context to get all the stats updated properly in the game. Do you know of one which works like the one for the character which doesn't require the party context?

Reply:
Ah, nevermind, i just realized it doesn't make sense to send a context if the pawn is not in the party, as the context is only ever used in parties. This approach is right.